### DISCUSSION TITLE
How to prevent ADH from assigning host names to clients?

### DISCUSSION THREAD

**Discussion by ghost** [AdguardTeam/AdGuardHome#3535] [Q&A]
Adguardhome enables DHCP and DNS services, How to prevent Adguardhome from assigning host names to clients.

**Comment by ameshkov** [comment]
Could you please explain why would you need that?

**Reply by ghost** [reply]
Because it assigns dynamic host names to my clients!

**Reply by ameshkov** [reply]
In theory, it should do that unless the client supplies its own hostname

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/51149477/132818385-92b6e38f-6b85-4b24-a3ca-5d2b73420a10.png)
ADG assigned me a host name (IP address). After the installation of the operating system, I tried to change the host name, but failed. After consulting the relevant data, the answer is to turn off DNS or DHCP to assign the host name to the client.

**Reply by ghost** [reply]
I have also encountered this problem. The DHCP service assign the host name to the client. If I try to turn off the DHCP service, this problem will not occur.

**Reply by ghost** [reply]
@ameshkov This problem occurs even if the host name is set, because the DHCP service assigns the client transient hostname instead of static hostname. However, it does not affect the use of the client.

**Reply by ainar-g** [reply]
@mingtdlb, we're currently in the process of refactoring our DHCP option logic.  But the problem that you have is probably due to the fact that your hostname contains a dot, while AGH expects it to only contain letters, numbers, and hyphens.

**Reply by ghost** [reply]
> 
> 
> @mingtdlb, we're currently in the process of refactoring our DHCP option logic. But the problem that you have is probably due to the fact that your hostname contains a dot, while AGH expects it to only contain letters, numbers, and hyphens.

No, it's not about characters. I have also encountered that the host name is the domain name inversely resolved from the IP (when the client IP is not a LAN IP such as 192.168.0.0/16, 10.0.0/8, 172.16.0.0/12).
